apps/public-docsite-v9/src/Concepts/Slots/Slots.stories.mdx

```diff
@@ -0,0 +1,350 @@
+Component children allow for building deep hierarchies using JSX/TSX.
+They also allow for heterogeneous types of content.
+
+For example, `Accordion` is a hierarchy of `AccordionItem` elements.
+
+```tsx
+<Accordion>
+  <AccordionItem value="1">
+    <AccordionHeader>Accordion Header 1</AccordionHeader>
+    <AccordionPanel>
+      <div>Accordion Panel 1</div>
+    </AccordionPanel>
+  </AccordionItem>
+  <AccordionItem value="2">
+    <AccordionHeader>Accordion Header 2</AccordionHeader>
+    <AccordionPanel>
+      <div>Accordion Panel 2</div>
+    </AccordionPanel>
+  </AccordionItem>
+  <AccordionItem value="3">
+    <AccordionHeader>Accordion Header 3</AccordionHeader>
+    <AccordionPanel>
+      <div>Accordion Panel 3</div>
+    </AccordionPanel>
+  </AccordionItem>
+</Accordion>
+```
+
+## Usage examples
+
+### Passing content
+
+You can pass a primitive value to slots.
+The `Input` component's `contentBefore` and `contentAfter` slots can be passed strings.
+
+```tsx
+<Input contentBefore="$" value="10" contentAfter=".00" />
+```
+
+You can pass JSX elements to slots. For example, the `Button` component can be passed an `img` or an `CalendarRegular24` icon.
+
+```tsx
+<>
+  <Button icon={<img src='site-icon.png' alt='branded site icon' /> }
+  <Button icon={<CalendarRegular24 />} />
+</>
+```
+
+### Passing props
+
+When a slot renders an element by default, you can pass props to it.
+
+The `Avatar` component has a `badge` slot that renders a `PresenceBadge`.
+This allows you pass `PresenceBadge` props.
+
+```tsx
+<Avatar name="Support" badge={{ status: 'available', 'aria-label': 'available' }} />
+```
+
+You can pass the `className` prop to customize the style of a slot.
+The `Avatar` badge slot supports a `className`, allowing the badge style to be customized.
+
+```tsx
+const useStyles = makeStyles({
+  badge: { color: tokens.colorBrandStroke1 },
+});
+
+const BusyBrandAvatar = () => {
+  const styles = useStyles();
+  return <Avatar name="IT probably" badge={{ status: 'busy', className: styles.badge }} />;
+};
+```
+
+You can pass the `as` prop to change the element type of a slot.
+Note that you must choose from one of the available element types the slot supports.
+
+```tsx
+<AccordionHeader as="h1">Accordion Header as h1</AccordionHeader>
+```
+
+### Replacing the entire slot
+
+When you pass content or props to a slot,
+the component renders the content within a component or element based on the slot type.
+If you need to replace the slot's entire content, including the containing element,
+pass a render function as the children.
+
+This is an escape hatch in the slots API, so prefer the other techniques whenever possible.
+
+By passing `renderBigLetterIcon` as the `children`, the `span` that normally contains the icon is replaced with an `h1`.
+
+```tsx
+const renderBigLetterIcon (Component, props) => {
+  return <h1>B<h1>;
+};
+
+<Button icon={{ children: renderBigLetterIcon }}>Bold</Button>;
```

Comment:
The result here is BBold. I think in the past you had 'old' on 158.  Example is also a bit odd because h1 is block level which would produce
B
Bold